商品削除機能

Gussan0829 opened the pull request:
# what
課題実装のGyazoの添付
ログイン状態の出品者のみ、詳細ページの削除ボタンを押すと、出品した商品を削除できる動画
https://gyazo.com/098f423b66c7660a5379b8626047aa3c
# why
課題実装のため

Gussan0829 committed:
コントローラーの設定

削除が完了したら、トップページに遷移すること。

Gussan0829 committed:
コントローラーの修正

Gussan0829 committed:
コントローラーの修正

mihot117 commented on app/controllers/items_controller.rb:
```diff
@@ -34,6 +34,12 @@ def update
     end
   end
 
+  def destroy
+    if @item.destroy
+      redirect_to  root_path
+    end
+  end
```
```ruby
    if @item.destroy
      redirect_to  root_path
    end
```

上記と合わせて、`出品者以外のユーザーが削除しようとした時`の対策も設定いただきますようお願いいたします。

削除機能の実装要件に以下のような記述ございます。

```ruby
「ログイン状態の場合にのみ、自身が出品した商品情報を削除できること」とは、詳細ページにおける「削除」ボタンの表示・非表示に加え、コントローラー側でも条件を設けることを指す。
```

自身が出品した商品のみ削除できるようにコントローラー側で設定し、
`出品者以外は削除できない`ように実装できれば要件達成となります。

destroyアクション内で条件を設定する場合は、
以下を参考に修正することで、コントローラー側でも正しい対策が可能となりますのでご参考ください。

```ruby
  def destroy
    if ログインユーザーが出品者である場合
      @item.destroy
      redirect_to root_path
    else
      render （削除に失敗したときの遷移先を設定）
    end
  end
```

上記の方法ではなく、`move_to_index`を使用することで、
コードをさらに整理することも可能です。
いずれか、実装しやすい方でお進みください。

mihot117 reviewed (commented):
レビュー依頼ありがとうございます。

コメントいたしましたので、ご確認をお願いいたします。

コードの修正後、コミット・プッシュを行った上で再度レビュー依頼をお送りください。
ご対応のほど、よろしくお願いいたします。

Gussan0829 committed:
コントローラーの修正

「ログイン状態の場合にのみ、自身が出品した商品情報を削除できること」とは、詳細ページにおける「削除」ボタンの表示・非表示に加え、コントローラー側でも条件を設けることを指す。

Gussan0829 commented:
# what
レビュー修正箇所の解消
# why
課題実装のため

chobi555 commented:
レビュー依頼ありがとうございます！
LGTMとさせていただきます、実装お疲れさまでした！

マージ後、ローカルのmasterブランチに忘れずにプルしましょう。
その後、新しいブランチを作成し次の実装に取り組んでいただけますと幸いです。
引き続き頑張っていきましょう！
